### PR TITLE
[file_selector_ios, image_picker_ios] Remove Swift Package Support

### PR DESCRIPTION
**Pull request by vashworth**
Plugins converted to Swift Package Manager that use a modulemap seem to be having issues with there being a `module.modulemap`. As a temporary solution to unblock people, we'll remove the `module.modulemap` and Swift Package support (`Package.swift`). 

Temporary solution for https://github.com/flutter/flutter/issues/148307.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by loic-sharma — APPROVED** [review]
LGTM

**Comment by stuartmorgan** [comment]
Do we have any idea yet why we don't see this in CI? We can land this as is, but it would be nice to have a test that we can ensure we'd don't regress later when re-landing support.

File: packages/file_selector/file_selector_ios/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.5.2+1
+
+* Removes Swift Package Manager compatibility.
```
Comment (stuartmorgan) [inline packages/file_selector/file_selector_ios/CHANGELOG.md:3]: How about "Temporarily remove Swift Package Manager compatibility to resolve issues with Cocoapods builds."?

**Comment by vashworth** [comment]
> Do we have any idea yet why we don't see this in CI? We can land this as is, but it would be nice to have a test that we can ensure we'd don't regress later when re-landing support.

So far, seems to be older versions of Xcode are having this issue and we don't enforce Xcode 15 yet. So unfortunately impossible to test in CI since macOS 13 does not support Xcode 14 or lower.

**Comment by stuartmorgan** [comment]
Have we actually validated that it's an Xcode 14 vs 15 thing? I thought that was still speculative based on the issue discussion (i.e., nobody seems to have indicated confirming that the same project works in 15 but not 14).

**Comment by vashworth** [comment]
> Have we actually validated that it's an Xcode 14 vs 15 thing? I thought that was still speculative based on the issue discussion (i.e., nobody seems to have indicated confirming that the same project works in 15 but not 14).

I confirmed Xcode 14 breaks: https://ci.chromium.org/ui/p/flutter/builders/try.shadow/Mac_x64%20ios_build_all_packages%20master/484/overview

**Review by jmagman — APPROVED** [review]

**Comment by vashworth** [comment]
> I think we can get rid of the manual modulemaps, that was a lot of work to `explicit module Test` working. We could just do the cheap thing and redeclare the methods we need in the tests.

Unfortunately, we're not the only ones that do this, though. I've seen other plugin authors copy our model to achieve the same. I would think we would want to provide a way for them to maintain it, but yeah the modulemaps are super annoying and I wish we could get rid of them. I'll think on it more.

Example: https://github.com/Baseflow/flutter-geolocator/blob/ca5469fe4212fe82d586bed93634d5e5a0e6154a/geolocator_apple/ios/Classes/GeolocatorPlugin.modulemap#L24

**Comment by stuartmorgan** [comment]
Have we considered cherry picking an Xcode 15 requirement to 3.22?

**Comment by vashworth** [comment]
> Have we considered cherry picking an Xcode 15 requirement to 3.22?

We could, but this issue isn't the only problem we're having with the double modulemaps. @loic-sharma also ran into an issue when trying to convert `camera_avfoundation`: https://gist.github.com/loic-sharma/27bd8e3ea0d9ab16cd4b8de03918ec4d

So I think we need to figure out a better way to handle modulemaps

**Pull request merged**